bgpd/bgp_vty.c

```diff
@@ -6788,6 +6788,40 @@ ALIAS_HIDDEN(
 	"Restart interval in minutes\n"
 	"Only give warning message when limit is exceeded\n")
 
+/* Allow local-preference for eBGP. */
+DEFUN(neighbor_local_preference_ebgp, neighbor_local_preference_ebgp_cmd,
+      "neighbor <A.B.C.D|X:X::X:X|WORD> ebgp-allow-local-preference",
+      NEIGHBOR_STR NEIGHBOR_ADDR_STR2
+      "Allow using LOCAL_PREF for eBGP sessions\n")
+{
+	int idx_peer = 1;
+	struct peer *peer;
+
+	peer = peer_and_group_lookup_vty(vty, argv[idx_peer]->arg);
+	if (!peer)
+		return CMD_WARNING_CONFIG_FAILED;
+
+	peer_flag_set(peer, PEER_FLAG_EBGP_ALLOW_LOCAL_PREF);
+
+	return CMD_SUCCESS;
+}
+
+DEFUN(no_neighbor_local_preference_ebgp, no_neighbor_local_preference_ebgp_cmd,
+      "no neighbor <A.B.C.D|X:X::X:X|WORD> ebgp-allow-local-preference",
+      NO_STR NEIGHBOR_STR NEIGHBOR_ADDR_STR2
+      "Allow using LOCAL_PREF for eBGP sessions\n")
+{
+	int idx_peer = 2;
+	struct peer *peer;
+
+	peer = peer_and_group_lookup_vty(vty, argv[idx_peer]->arg);
+	if (!peer)
+		return CMD_WARNING_CONFIG_FAILED;
+
```

Comment:
What happens for already advertised rateability info for this peer when we disable this command for EBGP?

Reply:
do you suggest sending a notification about the changed configuration?

Reply:
If you notify, it will reset the adj, correct? Which is the right course of action anyway...

Reply:
@Spantik @riw777 added notification after configuring/deconfiguring.